charts/kube-state-metrics/templates/role.yaml

```diff
@@ -1,180 +1,192 @@
-{{- if .Values.rbac.create -}}
+{{- if and (eq $.Values.rbac.create true) (not .Values.rbac.useExistingRole)  -}}
+{{- if eq .Values.rbac.useClusterRole false }}
+{{-   range (split "," $.Values.namespace) }}
+{{- end }}
+---
 apiVersion: rbac.authorization.k8s.io/v1
+{{- if eq .Values.rbac.useClusterRole false }}
+kind: Role
+{{- else }}
 kind: ClusterRole
+{{- end }}
 metadata:
   labels:
-    app.kubernetes.io/name: {{ template "kube-state-metrics.name" . }}
-    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-  name: {{ template "kube-state-metrics.fullname" . }}
+    app.kubernetes.io/name: {{ template "kube-state-metrics.name" $ }}
+    helm.sh/chart: {{ $.Chart.Name }}-{{ $.Chart.Version }}
```

Comment:
Hi @KlavsKlavsen, Can you elaborate on why we need to use the global scope explicitly here?

Reply:
. does't work when inside a range loop. $ achieves the same thing and works inside a loop.

Reply:
Oh yes, I am aware of using the $ scope within range. Is it the range from Line 3? I didn't quite understand that. I see an end in Line 4?

Reply:
I'm totally not understanding your question here.. sorry for being daft appearently :)
The end on line 4 - ends the IF - just before the range.. Its a bit peculiar - but the range is ONLY activated IF that if-clause right before is true. Its the only way I could make an optional range - seeing as I need to do either ClusterRole (once and no namespace(s) variable set) or otherwise a Role for each entry in namespace(s).